runtime/runtime-params-estimator/src/ext_costs_generator.rs

```diff
@@ -15,16 +15,25 @@ impl ExtCostsGenerator {
         Self { agg, result: Default::default() }
     }
 
-    fn extract_value(&mut self, metric: Metric, ext_cost: ExtCosts) -> Ratio<u64> {
-        let base = &self.agg[&Metric::noop];
-        let agg = &self.agg[&metric];
+    fn extract_value(&mut self, metric: Metric, ext_cost: ExtCosts) -> Option<Ratio<u64>> {
+        let base = self.agg.get(&Metric::noop)?;
+        let agg = self.agg.get(&metric)?;
         let multiplier = agg.ext_costs[&ext_cost];
-        Ratio::new(agg.upper_with_base(base), multiplier)
+        Some(Ratio::new(agg.upper_with_base(base), multiplier))
     }
 
     fn extract(&mut self, metric: Metric, ext_cost: ExtCosts) {
-        let gas = self.extract_value(metric, ext_cost);
-        self.result.insert(ext_cost, gas);
+        if let Some(value) = self.extract_value(metric, ext_cost) {
+            self.result.insert(ext_cost, value);
+        }
+    }
+
+    fn extract_max(&mut self, metric1: Metric, metric2: Metric, ext_cost: ExtCosts) {
+        if let Some(value1) = self.extract_value(metric1, ext_cost) {
```

Comment:
Shalln't we report error if one is present and another is not?

Reply:
I think it's fine to not report the error, as just we won't compute the cost then (that is, we won't return a wrong cost). That's essentially the same behavior we have for other complex costs computed out of several metrics -- if you omit the metric, the whole cost is omitted. 

The right behavior here is to just figure out what you need to compute to get the cost, and compute just that. That's a larger refactor that is being done #4733. The goal here is primaraly to be able to quickly test if #4733 returns roughly the same results.